code/game/gamemodes/cult/ritual.dm

```diff
@@ -82,762 +86,207 @@ var/engwords = list("travel", "blood", "join", "hell", "destroy", "technology",
-				if("emp")
-					R.word1=worddestr
-					R.word2=wordsee
-					R.word3=wordtech
-					R.loc = user.loc
-					R.check_icon()
-				if("drain")
-					R.word1=wordtravel
-					R.word2=wordblood
-					R.word3=wordself
-					R.loc = user.loc
-					R.check_icon()
-				if("raise")
-					R.word1=wordblood
-					R.word2=wordjoin
-					R.word3=wordhell
-					R.loc = user.loc
-					R.check_icon()
-				if("obscure")
-					R.word1=wordhide
-					R.word2=wordsee
-					R.word3=wordblood
-					R.loc = user.loc
-					R.check_icon()
-				if("astral journey")
-					R.word1=wordhell
-					R.word2=wordtravel
-					R.word3=wordself
-					R.loc = user.loc
-					R.check_icon()
-				if("manifest")
-					R.word1=wordblood
-					R.word2=wordsee
-					R.word3=wordtravel
-					R.loc = user.loc
-					R.check_icon()
-				if("imbue talisman")
-					R.word1=wordhell
-					R.word2=wordtech
-					R.word3=wordjoin
-					R.loc = user.loc
-					R.check_icon()
-				if("sacrifice")
-					R.word1=wordhell
-					R.word2=wordblood
-					R.word3=wordjoin
-					R.loc = user.loc
-					R.check_icon()
-				if("reveal")
-					R.word1=wordblood
-					R.word2=wordsee
-					R.word3=wordhide
-					R.loc = user.loc
-					R.check_icon()
-				if("wall")
-					R.word1=worddestr
-					R.word2=wordtravel
-					R.word3=wordself
-					R.loc = user.loc
-					R.check_icon()
-				if("freedom")
-					R.word1=wordtravel
-					R.word2=wordtech
-					R.word3=wordother
-					R.loc = user.loc
-					R.check_icon()
-				if("cultsummon")
-					R.word1=wordjoin
-					R.word2=wordother
-					R.word3=wordself
-					R.loc = user.loc
-					R.check_icon()
-				if("deafen")
-					R.word1=wordhide
-					R.word2=wordother
-					R.word3=wordsee
-					R.loc = user.loc
-					R.check_icon()
-				if("blind")
-					R.word1=worddestr
-					R.word2=wordsee
-					R.word3=wordother
-					R.loc = user.loc
-					R.check_icon()
-				if("bloodboil")
-					R.word1=worddestr
-					R.word2=wordsee
-					R.word3=wordblood
-					R.loc = user.loc
-					R.check_icon()
-				if("communicate")
-					R.word1=wordself
-					R.word2=wordother
-					R.word3=wordtech
-					R.loc = user.loc
-					R.check_icon()
-				if("stun")
-					R.word1=wordjoin
-					R.word2=wordhide
-					R.word3=wordtech
-					R.loc = user.loc
-					R.check_icon()
-				if("summon shell")
-					R.word1=wordtravel
-					R.word2=wordhell
-					R.word3=wordtech
-					R.loc = user.loc
-					R.check_icon()
+/obj/item/weapon/tome/proc/scribe_rune(mob/user)
+	var/chosen_keyword
+	var/rune_to_scribe
+	var/entered_grammar = stripped_input(user, "Enter the words of the rune you wish to scribe.", "Words of Power", "")
+	if(!entered_grammar)
+		return
+	for(var/T in typesof(/obj/effect/rune))
+		var/obj/effect/rune/R = new T(src)
+		if(R.grammar == entered_grammar)
+			rune_to_scribe = T
+		else
+			qdel(R)
+	if(!rune_to_scribe)
+		rune_to_scribe = /obj/effect/rune/malformed //If improper grammar is entered, the rune becomes malformed
+	if(rune_to_scribe == /obj/effect/rune/teleport)

```

Comment:
Teleport is the only rune that uses a keyword. If I tried to check for rune_to_scribe.keyword it would give me a compile error since the keyword variable doesn't exist. The only way it would work is if I typecasted rune_to_scribe is a teleport rune, which is bulkier and no different than what's in place already.